File: CS Degree/CS50/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // Get text from the user
    string text = get_string("Text: ");

    // Initialize letters, words and sentences counts
    int letters = 0;
    int words = 1;
    int sentences = 0;

    // Count letters, words and sentences
    for (int i = 0, characters = strlen(text); i < characters; i++)
    {
        int letterAscii = (int) text[i];
        if (letterAscii == 32)
        {
            words++;
        }
        else if (letterAscii == 33 || letterAscii == 46 || letterAscii == 63)
        {
            sentences++;
        }
        else if ((letterAscii >= 65 && letterAscii <= 90) || (letterAscii >= 97 && letterAscii <= 122))
        {
            letters++;
        }
    }

    // Print results
    // printf("%i letter(s)\n", letters);
    // printf("%i word(s)\n", words);
    // printf("%i sentence(s)\n", sentences);

    // Calculate Coleman-Liau index
    float averageLetters = (letters * 100.0) / words;
    float averageSentences = (sentences * 100.0) / words;
    int index = round((0.0588 * averageLetters) - (0.296 * averageSentences) - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: CS Degree/CS50/pset6/readability.py
text = input("Text: ")

# Initialize variables
letters = sentences = 0
words = 1

# Count letters, words and sentences
for i in range(len(text)):
    letterAscii = ord(text[i])
    if letterAscii == 32:
        words += 1
    elif letterAscii in [33, 46, 63]:
        sentences += 1
    elif (letterAscii in range(65, 91)) or (letterAscii in range(97, 123)):
        letters += 1

# Calculate Coleman-Liau index
averageLetters = letters * 100.0 / words
averageSentences = sentences * 100.0 / words
index = round((0.0588 * averageLetters) - (0.296 * averageSentences) - 15.8)

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print("Grade", index)
